components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/AccountConfirmationValidationHandler.java

```diff
@@ -128,12 +131,15 @@ public void handleEvent(Event event) throws IdentityEventException {
                 IdentityUtil.setIdentityErrorMsg(customErrorMessageContext);
                 throw new IdentityEventException(IdentityCoreConstants.USER_EMAIL_NOT_VERIFIED_ERROR_CODE,
                         "User : " + Utils.maskIfRequired(userName) + "'s email is not confirmed yet.");
-            } else if (!isSelfSignupEnabled && operationStatus && isUserEmailVerificationScenario(user)) {
-                IdentityErrorMsgContext customErrorMessageContext =
-                        new IdentityErrorMsgContext(IdentityCoreConstants.USER_EMAIL_NOT_VERIFIED_ERROR_CODE);
+            } else if (operationStatus && isUserEmailVerificationScenario(user)) {
```

Comment:
Have removed the `!isSelfSignupEnabled` check.
For email otp verification flow, it is not required to have the `isSelfSignupEnabled` check. This is added recently by this PR.
https://github.com/wso2-extensions/identity-governance/pull/989/files#diff-9835cfc5430a8fde95ccfe9295b958087762d8252a461463c7cf942f1914fd07R131

Reply:
Irrespective of self registration is enabled, email verification flow error code should be consistent.

Reply:
cc: @Malith-19

Reply:
Even re-triggering the email link based, it will be triggered as self-signup recovery scenario by the recovery portal (when self registration is enabled). 

That behavior is not ideally correct, but it won't happen for the email-otp verification case. It is corrected to always send the correct recovery scenario which is emai-otp-verification even for triggering.

The limitation disucssed here should be able to reproduce by triggering email-link verification with its corresponding recovery scenario. As per the existing logic, it won't fall into any condition. Will check and update this thread.